### PR TITLE
add export image button

### PR DESCRIPTION
**Pull request by Marmare314**
screenshot_2023-09-14_17-52-49

Only embedded images are allowed to be exported, but also embedded vanilla images.

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/editor.cpp [hunk 1]
```diff
@@ -785,6 +786,61 @@ bool CEditor::CallbackSaveCopyMap(const char *pFileName, int StorageType, void *
...
+		OutputFormat = IMAGE_FORMAT_RGB;
+		break;
+	case CImageInfo::FORMAT_RGBA:
+		OutputFormat = IMAGE_FORMAT_RGBA;
+		break;
+	case CImageInfo::FORMAT_SINGLE_COMPONENT:
+		OutputFormat = IMAGE_FORMAT_R;
+		break;
+	default:
+		pEditor->ShowFileDialogError("Image has invalid format.");
```
Comment (Robyt3) [inline src/game/editor/editor.cpp:818]: Should be impossible, so use `dbg_assert(false, "...")` instead.

**Comment by Robyt3** [comment]
There should be a popup with a message like "Make sure you have the permission to use these images and sounds in your own maps.", which is shown when using the `map_extract` tool. The popup should also show the author info like the tool if any is specified in the map.

**Comment by furo321** [comment]
The export button should probably also be available for sounds. Also I think the filename should be automatically filled to what the assets filename is.

**Comment by def-** [comment]
@Marmare314 Interested to fix up the last suggestions?

**Comment by heinrich5991** [comment]
I don't think such a popup is helpful. The same applies when saving an image from a website, e.g.. Appending a map doesn't show such a popup either.

**Pull request merged**